request.js

```diff
@@ -1,6 +1,15 @@
 var Request = (function () {
 
   var makeRequest = function(method, url, callback) {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
+      if (xhr.readyState === 4 && xhr.status === 200){
+        callback(JSON.parse(xhr.responseText));
```

Comment:
the callback should probably take 2 arguments (error, result), as was talked about yesterday